File: Typescript/Playwright.ts
// Requires: npm i playwright
const { chromium } = require('playwright');

const main = async () => {
  const browser = await chromium.launch({ headless: false });
  const page = await browser.newPage();

  await page.goto('https://example.com');
  console.log('Title:', await page.title());

  await page.screenshot({ path: 'example.png', fullPage: true });
  console.log('Screenshot saved as example.png');

  await browser.close();
};

main().catch((error) => {
  console.error('Unexpected error:', error);
});

File: Python/Playwright.py
from playwright.sync_api import sync_playwright


def main() -> None:
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=False)
        page = browser.new_page()

        page.goto("https://example.com")
        print("Title:", page.title())

        page.screenshot(path="example.png", full_page=True)
        print("Screenshot saved as example.png")

        browser.close()


if __name__ == "__main__":
    main()
